clash-lib/src/Clash/Netlist/BlackBox/Util.hs

```diff
@@ -474,10 +475,19 @@ renderElem b (IF c t f) = do
                       _            -> 0
 
       (IsUndefined n) ->
-        let (e, _, _) = bbInputs b !! n in
-        case (xOpt, e) of
-          (True, BlackBoxE _ _ _ _ (N.BBTemplate [Err _]) _ _) -> 1
-          _ -> 0
+        let
+          (e, _, _) = bbInputs b !! n
+          checkUndefined e0 = case e0 of
```

Comment:
Perhaps

```suggestion
          checkUndefined = \case
```

Reply:
I think `checkUndefined` is a very nice function to have on top-level too! I'm sure it'll come in handy somewhere else in the future.

Reply:
Done!